src/components/hooks/usePeoplePage.ts

```diff
@@ -0,0 +1,99 @@
+import React, {
+  useCallback, useEffect, useMemo, useState,
+} from 'react';
+import { useHistory, useLocation } from 'react-router-dom';
+import { getPeople } from '../../api/data';
+import debounce from '../../helpers/debounce';
+
+export const usePeoplePage = () => {
+  const history = useHistory();
+  const location = useLocation();
+  const search = new URLSearchParams(location.search);
+  const query = search.get('query') || '';
+  const sortedBy = search.get('sortBy');
+  const sortOrder = search.get('sortOrder');
+
+  const [people, setPeople] = useState([]);
+  const [inputValue, setInputValue] = useState(query);
+
+  useEffect(() => {
+    const fetchPreparedPeople = async () => {
+      const peopleFromServer = await getPeople();
+
+      setPeople(peopleFromServer.map((person: Person, i: number) => ({
+        ...person,
+        id: i + 1,
+        age: person.died as number - (person.born as number),
+        century: Math.ceil(person.died as number / 100),
+        mother: person.mother ? person.mother : '----',
+        father: person.father ? person.father : '----',
+        children: peopleFromServer
+          .filter((child: Person) => child.father === person.name || child.mother === person.name)
+          .map((child: Person) => child.name)
+          .join(', ') || '----',
+      })));
+    };
+
+    fetchPreparedPeople();
+  }, []);
+
+  const historyPush = debounce(() => {
+    history.push({ search: search.toString() });
+  }, 500);
+
+  const searchPeople = useCallback(
+    (e: React.ChangeEvent<HTMLInputElement>) => {
+      const value = e.target.value.trim();
+
+      search.set('query', value.toLowerCase());
+
+      if (!(search.get('query') || '').trim()) {
+        search.delete('query');
+      }
+
+      setInputValue(value);
+      historyPush();
+    }, [search, historyPush],
+  );
+
+  let searchedPeople: Person[] = useMemo(() => {
+    return people.filter(({ name, mother, father }: Person) => (
+      name as string + mother as string + father
+    ).toLowerCase().includes(query));
+  }, [people, query]);
+
+  const sortPeople = (arr: Person[], thead: string, order: string | null) => {
+    const sortedArr = [...arr].sort((a, b) => (
+      typeof a[thead] === 'string'
+        ? (a[thead] as string).localeCompare(b[thead] as string)
+        : (a[thead] as number) - (b[thead] as number)
+    ));
+
+    return order === 'desc' ? sortedArr.reverse() : sortedArr;
```

Comment:
I recommend you to make another sorting for desc order because you'll have twice more operations than necessary, optimize the code